### PR TITLE
optimize cdraw

### PR DESCRIPTION
**Pull request by tsedan**

### CODE REVIEW

**Comment by tsedan** [comment]
Converted to draft because there's some linux malloc issue i need to diagnose and I also have a few ideas for further optimization

File: rubato/c_src/cdraw.cpp [hunk 1]
```diff
@@ -47,39 +46,37 @@ PIXEL FUNCTIONS
 inline void setPixel(size_t _pixels, int width, int height, int x, int y, size_t color, bool blending = false) {
     if (x < width && y < height && x >= 0 && y >= 0) {
         int off = y * width + x;
-        uint32_t added = (uint32_t) color;
-        uint32_t* pixels = (uint32_t*) _pixels;
+        uint_fast32_t added = (uint_fast32_t) color;
+        uint_fast32_t* pixels = (uint_fast32_t*) _pixels;
```
Comment (boazddog) [inline rubato/c_src/cdraw.cpp]: this is wrong. uint_fast32_t is `at least 32 bits`, and you need `exactly 32 bits`. that is the only difference.

File: rubato/c_src/cdraw.cpp [hunk 2]
```diff
@@ -467,17 +461,16 @@ inline void _aaDrawPoly(size_t _pixels, int width, int height, void* vx, void* v
 
 // Fill a polygon with the specified color.
 inline void _fillPolyConvex(size_t _pixels, int width, int height, void* vx, void* vy, int len, size_t color, bool blending) {
+    int* v_x_min = (int*) malloc(height * sizeof(int));
```
Comment (boazddog) [inline rubato/c_src/cdraw.cpp:464]: sizeof(int32_t)?

File: rubato/struct/gameobject/physics/hitbox.py [hunk 3]
```diff
@@ -259,8 +255,8 @@ def redraw(self):
         verts = [v + Vector(w // 2, h // 2) for v in self.verts]
 
         if self.color is not None:
-            self._image.draw_poly(verts, fill=self.color, aa=True)
-        self._debug_image.draw_poly(verts, Color.debug, 2)
+            self._image.draw_poly(verts, fill=self.color, aa=True, blending=False)
+        self._debug_image.draw_poly(verts, Color.debug, 2, blending=False)
```
Comment (tinmarr) [inline rubato/struct/gameobject/physics/hitbox.py:259]: Why turn blending off?

**Review by tinmarr — APPROVED** [review]
Just readded the rgba32 tests to get 100% coverage and merged the from_0 tests with the regular tests because that was an old and much worse testing convention.

**Pull request merged**